### PR TITLE
Run openstack-ardana jobs on some automation PRs

### PR DESCRIPTION
**Pull request by jesusaurus**
Trigger an openstack-ardana job in jenkins for pull requests to the automation repo if the change touches code paths used by the openstack-ardana job (self-gating).

Jira: SCRD-4307

### CODE REVIEW

File: scripts/github_pr/github_pr_cloud.yaml [hunk 1]
```diff
@@ -46,6 +47,26 @@ template:
...
+          - type: JenkinsJobTriggerArdana
+            parameters:
+              detail_logging: true
+              job_name: openstack-ardana
+              job_cmd: "../jenkins/jenkins-job-trigger"
+              job_parameters:
+                <<: *ardana_automation_parameters
+        blacklist_handler:
+          - type: SetStatus
+            config:
```
Comment (jesusaurus) [inline scripts/github_pr/github_pr_cloud.yaml]: @jdsn my ruby is not very strong, will setting the config.context object here automagically work, or will setting a unique context value per SetStatusAction require changes to github-pr?

File: jenkins/ci.suse.de/pipelines/openstack-ardana.Jenkinsfile [hunk 2]
```diff
@@ -406,23 +405,26 @@ pipeline {
...
         if [ -n "$github_pr" ] ; then
-          automation-git/scripts/ardana/pr-failure.sh
+          scripts/jenkins/ardana/pr-failure.sh
         else
-          automation-git/scripts/jtsync/jtsync.rb --ci suse --job $JOB_NAME 1
+          scripts/jtsync/jtsync.rb --ci suse --job $JOB_NAME 1
         fi
       '''
     }
+    cleanup {
```
[the remark this answers is not in the document]
Comment (jesusaurus) [reply jenkins/ci.suse.de/pipelines/openstack-ardana.Jenkinsfile]: I found the "condition" listed at https://jenkins.io/doc/book/pipeline/syntax/#post, and yeah, I can write up another change to update that most places

File: scripts/jenkins/ardana/pr-update.sh [hunk 3]
```diff
@@ -22,5 +22,5 @@ $ghpr --action set-status $ghpr_paras --status "pending" --targeturl $BUILD_URL
 git config --add remote.origin.fetch "+refs/pull/*/head:refs/remotes/origin/pr/*"
 git fetch origin $github_pr_sha
 git checkout -B ardana-ci FETCH_HEAD
-git clean  # remove files deleted from git
+git clean -f  # remove files deleted from git
```
Comment (jdsn) [inline scripts/jenkins/ardana/pr-update.sh]: This fix should be in a separate commit.

File: scripts/github_pr/github_pr_cloud.yaml [hunk 4]
```diff
@@ -46,6 +47,27 @@ template:
...
+              context: suse/ardana
```
[the remark this answers is not in the document]
Comment (jesusaurus) [reply scripts/github_pr/github_pr_cloud.yaml]: If we follow the context-string model you have described, then how can we make the ardana job required for changes that affect ardana? If one job sets `suse/mkcloud` and another sets `suse/mkcloud/ardana`, then can we make both contexts required?

I was attempting to put them side-by-side so that they could both be required (and set them successful when the jobs are skipped) and so that it would be obvious to the developer which jobs to look at after a failure.

File: scripts/github_pr/github_pr_cloud.yaml [hunk 5]
```diff
@@ -46,6 +47,27 @@ template:
...
+            parameters:
+              status: success
+              message: Ardana gating not applicable
```
[the remark this answers is not in the document]
Comment (jesusaurus) [reply scripts/github_pr/github_pr_cloud.yaml]: The status message is set per-context, so this would only report back to the `suse/ardana` (or `suse/mkcloud/ardana`) status.

**Comment by dirkmueller** [comment]
@jesusaurus it looks like you're not going to work on this any time soon, please don't hesitate to reopen the PR (instead of creating a new one) should this change.
